src/cpuinfo/SDL_cpuinfo.c

```diff
@@ -1121,11 +1147,11 @@ SDL_SIMDRealloc(void *mem, const size_t len)
         if (memdiff != ptrdiff) { /* Delta has changed, copy to new offset! */
             oldmem = (void*) (((uintptr_t) ptr) + memdiff);
 
-            /* Even though the data past the old `len` is undefined, this is the
+            /* Even though the data past the old `size` is undefined, this is the
              * only length value we have, and it guarantees that we copy all the
              * previous memory anyhow.
              */
-            SDL_memmove(retval, oldmem, len);
+            SDL_memmove(retval, oldmem, size);
```

Comment:
I think this is actually undefined behaviour (reading outside the bounds of an object is UB). Obviously `SDL_SIMDRealloc()` already exists, so SDL can't avoid that; but perhaps it would be better to limit the damage by not adding `SDL_aligned_realloc()`, and possibly also deprecating `SDL_SIMDRealloc()`?

(The way to avoid the UB here would be to either store the size of the allocation similar to the way the allocated pointer is stored, or use non-portable out-of-band information like glibc `malloc_usable_size()`.)

It seems the [only caller](https://codesearch.debian.net/search?q=SDL_SIMDRealloc&literal=1) of `SDL_SIMDRealloc()` visible in Debian's code search is `pysdl2`, which binds it out of a sense of completeness, but honestly probably shouldn't).

Reply:
> by not adding `SDL_aligned_realloc()`

... or at least not adding it as public API (having it available internally so that `SDL_SIMDRealloc()` can call it would make sense).

Reply:
Yeah, we shouldn't have SDL_aligned_realloc(). SDL_SIMDRealloc() can potentially be a stub that returns NULL and is deprecated in 2.24.0.

Reply:
I haven't looked carefully at the detail of implementation. But doing a SIMDRealloc should be safe:
get the initial pointer. SDL_Realloc new_size + alignment. do a correct/tricky memmove on new pointer to have the real data on the new alignment + the real data at place 0.
There should be no out of bounds / sigsev or Undefined Behavior. 
Only the memmove will most of the time move un-initialized memory (end of new area). but this isn't a UB.

It's rare to use SIMDAlloc, and even more rare to use SIMDRealloc. But this sounds legitimate to have this functions.
Let's say you have an audio buffer where you doing simd processing, and you want to reallocated it. then you can use SDL_SIMDRealloc.

But .... SDL_aligned_realloc is probably broken. because we dont save the alignement.
so if the first alignement of SDL_aligned_realloc is different from the second alignement of SDL_aligned_realloc. then we don't do the correct SDL_memmove. (Unless we store the alignement in ptr[-2] for instance ...). 
So I'm going to remove SDL_aligned_realloc...